README.md

```diff
@@ -15,7 +72,342 @@ def deps do
+
+  1. **Create your BlockType module**: Continuing from the example scenario outlined above, create a new block type at `lib/your_app/block_types/image.ex` where `image` is just an example of a descriptive name of the block you're modeling.
+
+      ```elixir
+      # lib/your_app/block_types/image.ex
+
+      defmodule YourApp.BlockTypes.ImageBlock do
+        # 1. use the BlockType macro, and specify the type string that matches the data returned by your CKJS block type
+        use ColonelKurtz.BlockType, type: "image"
+
+        # 2. use the `defattributes` macro to specify the schema for your block's content
+        defattributes(
+          src: :string,
+          width: :integer,
+          height: :integer
+        )
+
+        # 3. optional, but encouraged - define your validation rules for the block's content
+        def validate_content(_content, changeset) do
+          changeset
+          |> validate_required([:src, :width, :height])
+          # ... any other custom validation rules you need ...
+        end
+
+        # 4. optional, as necessary - define validation rules for the block itself
+        def validate(_block, changeset) do
+          changeset
+          # e.g. this block must have at least 1 child block
+          |> validate_length(:blocks, min: 1)
+        end
+      end
+      ```
+
+  2. **Create your BlockView**: create a new block view at `lib/your_app_web/blocks/image_view.ex`.
+
+      **Note:** make sure you've configured `ColonelKurtzEx` with the location of your `block_views` and `block_types`. See "**Configure `ColonelKurtzEx`**" above.
+
+      ```elixir
+      # lib/your_app_web/blocks/image_view.ex
+
+      defmodule YourAppWeb.Blocks.ImageView do
+        use YourAppWeb, :view
+        use ColonelKurtz.BlockTypeView
+
+        # optionally implement `renderable?/1`
+        def renderable?(%ImageBlock{content: %{src: ""}} = block), do: false
+        def renderable?(_block), do: true
+      end
+      ```
+  ---
+</details>
+
+## FAQs
+
+<details>
+  <summary>
+    <strong>
+      How does <code>ColonelKurtzEx</code> look up my custom block type and view modules?
+    </strong>
+  </summary>
+
+  You configure the `:block_views` and `:block_types` options for `CKEX` in your `config.exs` by providing the modules in your app that will contain your custom block types and views. When `CKEX` marshalls blocks JSON, it parses data into lists of maps using `Jason` and then looks up a block type based on the block's `type` field.
+
+  It does so by calling `Module.concat` with the module you specified for `:block_types` and `Macro.camelize(type) <> "Block"` (e.g. `"image"` => `YourApp.BlockTypes.ImageBlock`).
+
+  Similarly, to lookup your view modules `CKEX` calls `Module.concat` with the module you specified for `:block_views` and `Macro.camelize(type) <> "View"` (e.g. `"image"` => `YourAppWeb.Blocks.ImageView`).
```

Comment:
It's awesome that you documented this well, but this functionally, this part is definitely confusing. Feels more OO than functional.

Reply:
Hmm.. I'm definitely open to suggestions for better ways of doing this. We get data that has a `type: :string` field from CK JSON and we need to match that up with a module in order to call the right changeset function and cast data properly. We're doing a little runtime dynamic module resolution in order to do that. An alternative might be having some kind of stateful registry that get's populated with a mapping of type to module and has a public api for looking up the module given a type. So far we've avoided having anything stateful in the library (and doing runtime module lookups lets us do that).